lib/perl/Genome/VariantReporting/Suite/BamReadcount/PerLibraryVafInterpreter.t

```diff
@@ -0,0 +1,158 @@
+            'Solexa-135852_ref_count' => '2',
+            'Solexa-135853_ref_count' => '1',
+            'Solexa-135852_vaf' => '87.0786516853933',
+            'Solexa-135853_vaf' => '99.4652406417112',
+        }
+    );
+
+    my $entry = create_default_entry();
+    my %result = $interpreter->interpret_entry($entry, ['G']);
+    is_deeply(\%result, \%expected, "Values are as expected");
+};
+
+subtest "insertion" => sub {
+    my $interpreter = $pkg->create(
+        sample_names => ["S4"],
+        library_names => $library_names
+    );
+    lives_ok(sub {$interpreter->validate}, "Interpreter validates");
+
+    my %expected = (
+        AA => {
+            'Solexa-135852_var_count' => '20',
+            'Solexa-135853_var_count' => '0',
+            'Solexa-135852_ref_count' => '2',
+            'Solexa-135853_ref_count' => '1',
+            'Solexa-135852_vaf' => '11.2359550561798',
+            'Solexa-135853_vaf' => '0',
+        }
+    );
+
+    my $entry = create_default_entry();
+    my %result = $interpreter->interpret_entry($entry, ['AA']);
+    is_deeply(\%result, \%expected, "Values are as expected");
+};
+
+subtest "deletion" => sub {
+    my $interpreter = $pkg->create(
+        sample_names => ["S1"],
+        library_names => $library_names
+    );
+    lives_ok(sub {$interpreter->validate}, "Interpreter validates");
+
+    my %expected = (
+        A => {
+            'Solexa-135852_var_count' => '20',
+            'Solexa-135853_var_count' => '0',
+            'Solexa-135852_ref_count' => '3',
+            'Solexa-135853_ref_count' => '2',
+            'Solexa-135852_vaf' => '11.0497237569061',
+            'Solexa-135853_vaf' => '0',
+        }
+    );
+
+    my $entry = create_deletion_entry();
+    my %result = $interpreter->interpret_entry($entry, ['A']);
+    is_deeply(\%result, \%expected, "Values are as expected");
+};
+
+subtest "long indel" => sub {
+    my $interpreter = $pkg->create(
+        sample_names => ["H_KA-174556-1309245"],
+        library_names => [qw(H_KA-174556-1309245-lg3-lib1 H_KA-174556-1309245-lg5-lib1)]
+    );
+    lives_ok(sub {$interpreter->validate}, "Interpreter validates");
+
+    my %expected = (
+        'GTATA' => {
+            'H_KA-174556-1309245-lg3-lib1_ref_count' => 29,
+            'H_KA-174556-1309245-lg3-lib1_vaf' => 6.25,
+            'H_KA-174556-1309245-lg3-lib1_var_count' => 2,
+            'H_KA-174556-1309245-lg5-lib1_ref_count' => 18,
+            'H_KA-174556-1309245-lg5-lib1_vaf' => 9.52380952380952,
+            'H_KA-174556-1309245-lg5-lib1_var_count' => 2,
+        }
+    );
+
+    my $entry = create_long_deletion_entry();
+    my %result = $interpreter->interpret_entry($entry, ['GTATA']);
+    is_deeply(\%result, \%expected, "Values are as expected");
+};
+
+subtest "libraries don't match" => sub {
+    for my $incorrect_library_names ([qw(Solexa-135853)], [qw(Solexa-135854 Solexa-135853 Solexa-135852)], [qw(nonexistent_library)]) {
```

Comment:
It could be but the way that the interpreter was originally written it will always return the vaf fields for all of the sample's libraries. When the library_names were added they were only ever used to  determine that available fields. We never restricted the return values to those libraries only. So I added this check to bring the available fields  in line with the actual output of this interpreter. 

I'm totally open to changing this behavior but I think it would need to happen in a different PR.